### PR TITLE
Update wso2is_analytics module for minimum HA deployment

### LINKED ISSUE #37: Refactor puppet artifacts to match with standard IAM deployment patterns of IS 5.4.0
**Description:**
Puppet artifacts should adhere to following standard IAM deployment patterns.
Pattern-1: HA clustered deployment of WSO2 Identity Server
Pattern-2: HA clustered deployment of WSO2 Identity Server with WSO2 Identity Analytics

**Affected Product Version:**
IS 5.4.0

### PR DESCRIPTION
**Pull request by malithie**
## Purpose
Resolves #37

## Goals
Update wso2is_analytics module for minimum HA deployment

## Approach
- Add datasources for event store and processed data store
- Enable clustering and ha deployment
- Removed unnecessary hiera entries
- Update readmes accordingly

## Test environment
JDK 8
Ubuntu 14.04
Puppet 3.8.7
Tested in puppet master agent setup

### CODE REVIEW

**Review by DilanUA — APPROVED** [review]

File: wso2is/hieradata/dev/wso2/wso2is/pattern-1/README.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # WSO2 Identity Server Pattern-1
 
-WSO2 Identity Server Deployment Pattern 1 runs two Identity Server instances fronted with a load balancer. Identity 
+[WSO2 Identity Server Deployment Pattern 1](https://docs.wso2.com/display/IS540/Deployment+Patterns#DeploymentPatterns-Pattern1-HAclustereddeploymentofWSO2IdentityServer) 
+runs two Identity Server instances fronted with a load balancer. Identity
```
Comment (DilanUA) [inline wso2is/hieradata/dev/wso2/wso2is/pattern-1/README.md]: I do not think that we deal with LBs at this level. So, do we need to mention that here ?

Comment (malithie) [reply wso2is/hieradata/dev/wso2/wso2is/pattern-1/README.md]: Well, this is added to give a brief explanation on the deployment pattern that the respective hiera of the module is referring. Yes we don't need to talk about load balancers. But, a person should know which component in which deployment pattern can be configured over this hiera file.
So if you just go below on the same readme you can see, that after explaining the deployment pattern we have explained which component of this deployment pattern can be configured with the hiera yaml file.
Quoting:
"The hiera YAML in this directory can be used to setup each WSO2 Identity Server node participating in the cluster."

Comment (DilanUA) [reply wso2is/hieradata/dev/wso2/wso2is/pattern-1/README.md]: When you say "Deployment Pattern 1 runs two Identity Server instances fronted with a load balancer", that might imply that we configure a LB as well here which is not the case, so let's just change the wording here.

Comment (malithie) [reply wso2is/hieradata/dev/wso2/wso2is/pattern-1/README.md]: I don't agree. It clearly says the use of hiera yaml file below.
Also, if you go to the root readme there is a clear indication of what can be acheived over this repository artifacts.
Quoting:
"Note that this guide, explains only about setting up wso2is or wso2is_analytics puppet modules with respect to the deployment pattern selected. Setting up the infrastructure, load balancers and sharing runtime deployment artifacts among the nodes participating in a deployment is not in the scope of this guideline. Refer WSO2 Identity Server Deployment Guidelines to understand how to do a complete deployment of WSO2 Identity Server based on a standard deployment pattern."

Comment (DilanUA) [reply wso2is/hieradata/dev/wso2/wso2is/pattern-1/README.md]: Make sense. +1

**Pull request merged**